api/server/graph.cc

```diff
@@ -70,19 +70,12 @@ Graph::~Graph(void) {
 
 bool Graph::LinkAndStalk(Graph::BrickShrPtr westBrick,
                          Graph::BrickShrPtr eastBrick,
-                         Graph::BrickShrPtr sniffer) {
-    if (app::config.packet_trace) {
-        if (pg_brick_chained_links(&app::pg_error, westBrick.get(),
-                                   sniffer.get(), eastBrick.get()) < 0) {
-            PG_ERROR_(app::pg_error);
-            return false;
-        }
+                         Graph::BrickShrPtr sniffer, bool enable) {
+    if (enable) {
+        link(westBrick, sniffer);
```

Comment:
sadly, I don't think that's okay to use link

You need to use link, when updating tracing,  but not when crating a nic.
The code work, but as Graph::LinkAndStalk is used inside nic add, by doing that, you slow down pooling more that it should at every nic add.

So either you give link function pointer in parameter, so you can configure the way linkAndStalk will link, either you need to create a new linkAndStaklk.